core/src/main/java/org/elasticsearch/action/search/SearchResponseListenerRegistry.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.action.search;
+
+
+import org.elasticsearch.plugins.SearchPlugin;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class SearchResponseListenerRegistry {
```

Comment:
Happy to delete this whole thing if there's a better/easier way :)

I thought `SearchOperationListener` was only available on a per-shard basis, being retrieved from the `IndexShard` for that particular shard?  If there's a way to grab the `SearchOperationListener` in the final AsyncAction it'd be great, I just couldn't see a way to do it at the time.

(that's why I didn't use the onQueryPhase hook that's currently in place, it's per-shard)